wsmaster/codenvy-hosted-system-ram/src/main/java/com/codenvy/service/systemram/DockerBasedSystemRamInfoProvider.java

```diff
@@ -0,0 +1,120 @@
+package com.codenvy.service.systemram;
+
+import com.google.inject.Inject;
+
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.SystemInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.eclipse.che.commons.lang.Size.parseSize;
+
+/**
+ * Implementation of {@link SystemRamInfoProvider} based on docker.
+ *
+ * @author Igor Vinokur
+ */
+public class DockerBasedSystemRamInfoProvider implements SystemRamInfoProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DockerBasedSystemRamInfoProvider.class);
+
+    private static final String SYSTEM_RAM_INFO_ERROR = "An error occurred while getting system RAM info.";
+
+    private final DockerConnector dockerConnector;
+
+    @Inject
+    public DockerBasedSystemRamInfoProvider(DockerConnector dockerConnector) {
+        this.dockerConnector = dockerConnector;
+    }
+
+    @Override
+    public SystemRamInfo getSystemRamInfo() throws ServerException {
+        long systemRamUsed = 0;
+        long systemRamTotal = 0;
+        SystemInfo systemInfo;
+        List<String> allNodesRamUsage = new ArrayList<>();
+
+        try {
+            systemInfo = dockerConnector.getSystemInfo();
+        } catch (IOException e) {
+            LOG.error("Failed to retrieve system information from docker.", e);
+            throw new ServerException(SYSTEM_RAM_INFO_ERROR, e);
+        }
+        String[][] systemValues = systemInfo.getDriverStatus() == null ? systemInfo.getSystemStatus() : systemInfo.getDriverStatus();
+        if (systemValues == null) {
+            LOG.error("Empty system information was received from docker. All system info from docker {}", systemInfo);
+            throw new ServerException(SYSTEM_RAM_INFO_ERROR);
+        }
+
+        /*
+          system values from docker response are introduced in a 2-dimensional string array e.g.:
+          "SystemStatus": [
+             .
+             .
+             .
+             " node1.<host>",
+             "<host>:<port>"
+             ],
+             "  └ Reserved Memory",
+             "1000 MiB / 5.881 GiB"
+             ],
+             .
+             .
+             .
+             " node2.<host>",
+             "<host>:<port>"
+             ],
+             [
+             "  └ Reserved Memory",
+             "1000 MiB / 1.511 GiB"
+             ],
+             .
+             .
+             .
+          ],
+          RAM vales are divided by nodes and introduced in a string array which has '└ Reserved Memory' in first element,
+          and RAM values of node in format <used RAM size> / <total RAM size> in second element.
+         */
+        for (String[] systemInfoEntry : systemValues) {
+            if (systemInfoEntry.length == 2 && " └ Reserved Memory".equals(systemInfoEntry[0])) {
```

Comment:
It can be more than one time, if more then one node is present in the system. Added this to description.